samples/dotnetcore/image-transfer/ContainerRegistryTransfer/Clients/ExportClient.cs

```diff
@@ -0,0 +1,125 @@
+                    Type = "AzureStorageBlobContainer",
+                    Uri = options.ExportPipeline.ContainerUri,
+                    KeyVaultUri = options.ExportPipeline.KeyVaultUri
+                },
+                options: options.ExportPipeline.Options
+                );
+
+                return registryClient.ExportPipelines.Create(registryName: options.ExportPipeline.RegistryName,
+                                                                resourceGroupName: options.ExportPipeline.ResourceGroupName,
+                                                                exportPipelineName: options.ExportPipeline.PipelineName,
+                                                                exportPipelineCreateParameters: exportPipeline);
+            }
+            else
+            {
+                throw new ArgumentException($"Could not find registry '{exportRegistryName}'. Please ensure the registry exists in the current resource group {exportResourceGroupName}.");
+            }
+        }
+
+        public async Task ExportImagesAsync(ExportPipeline exportPipeline)
+        {
+            var pipelineId = exportPipeline.Id;
+            var pipelineRunName = options.ExportPipelineRun.PipelineRunName;
+            var targetName = options.ExportPipelineRun.TargetName;
+            var artifacts = options.ExportPipelineRun.Artifacts;
+
+            Console.WriteLine($"Export PipelineRun;");
+            Console.WriteLine($"  registryName: {options.ExportPipeline.RegistryName}");
+            Console.WriteLine($"  pipelineRunName: {options.ExportPipelineRun.PipelineRunName}");
+            Console.WriteLine($"  pipelineResourceId: {pipelineId}");
+            Console.WriteLine($"  targetName: {options.ExportPipelineRun.TargetName}");
+            Console.WriteLine($"  artifacts: {string.Join("\n", artifacts)}");
+            Console.WriteLine($"======================================================================");
+
+
+            var pipelineRunRequest = new PipelineRunRequest
+            {
+                PipelineResourceId = pipelineId,
+                Target = new PipelineRunTargetProperties
+                {
+                    Type = "AzureStorageBlob",
+                    Name = targetName
+                },
+                Artifacts = artifacts
+            };
+
+            Console.WriteLine($"Running pipelineRun {pipelineRunName}...");
+
+            var pipelineRun = await registryClient.PipelineRuns.CreateAsync(registryName: options.ExportPipeline.RegistryName,
+                                                            resourceGroupName: options.ExportPipeline.ResourceGroupName,
+                                                            pipelineRunName: pipelineRunName,
+                                                            request: pipelineRunRequest).ConfigureAwait(false);
+            if (string.Equals(pipelineRun.ProvisioningState, "Failed", StringComparison.InvariantCultureIgnoreCase))
```

Comment:
>InvariantCultureIgnoreCase [](start = 88, length = 26)

The code is dealing with response from service not user input from UI. Usually it's better to use OriginalIgnoreCase.

Reply:
Good point, I typically always default to invariant culture but it is not needed here.